find/scripts/extract_download_logs.py

```diff
@@ -72,33 +70,42 @@ def extract_ids_from_message(message: List[List[dict]]) -> List[str]:
     return account_ids
 
 
-def fetch_user_data(user_ids: List[str]) -> dict:
+def get_email_addresses_for_user_ids(user_ids: List[str]) -> dict:
     """
-    Fetch user data from account-store API, using users' azure_ad_subject_id-s
+    Fetch user data from account-store API, using users' account_id-s
     args: user_ids: list of user_ids
     returns: dict of user_id: email_address
     """
+
+    account_store_api_host = os.getenv("ACCOUNT_STORE_API_HOST", "http://localhost:3003")
     responses = {}
-    for user_id in user_ids:
-        try:
-            response = requests.get(
-                f"{Config.ACCOUNT_STORE_API_HOST}/accounts",
-                headers={"Content-Type": "application/json"},
-                params={"azure_ad_subject_id": user_id},
-            )
-            response.raise_for_status()
-            response_data = response.json()
-            responses[user_id] = response_data["email_address"]
-        except requests.exceptions.RequestException as e:
-            print(f"An error occurred for user_id {user_id}: {e}")
+    try:
+        url_params = ",".join(user_ids)
+        url = f"{account_store_api_host}/bulk-accounts?account_id={url_params}"
+        response = requests.get(
+            url,
+            headers={"Content-Type": "application/json"},
+        )
+        response.raise_for_status()
+        response_data = response.json()
+
+        for user_id in user_ids:
+            if user_id in response_data:
+                responses[user_id] = response_data[user_id]["email_address"]
+            else:
+                responses[user_id] = None
```

Comment:
Can we represent this more concisely with a list comprehension e.g.
```suggestion
responses = {user_id: response_data.get(user_id, {}).get("email_address") for user_id in user_ids}
```

Reply:
Changed it as requested.